### PR TITLE
refactor(detected_object_validation): rework parameters

### PR DESCRIPTION
**Pull request by batuhanbeytekin**
## Description
Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371

## Tests performed
Package built and launch locally.

## Effects on system behavior
More reliable and faster parameter configuration file creation.

## Pre-review checklist for the PR author
The PR author **must** check the checkboxes below when creating the PR.
- [x] I've confirmed the [contribution guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/).
- [x] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).

## In-review checklist for the PR reviewers
The PR reviewers **must** check the checkboxes below before approval.
- [ ] The PR has been properly tested.
- [ ] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author
The PR author **must** check the checkboxes below before merging.
- [x] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: perception/detected_object_validation/schema/object_lanelet_filter.schema.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  "definitions": {
+    "object_lanelet_filter": {
+      "type": "object",
+      "properties": {
+        "filter_target_label": {
+          "type": "object",
+          "properties": {
+            "UNKNOWN": {
+              "type": "boolean",
+              "default": false,
```
Comment (badai-nguyen) [inline perception/detected_object_validation/schema/object_lanelet_filter.schema.json:14]: @batuhanbeytekin Thank you for your PR. Did you have any intention to change the default parameter here, compared with param.yaml?
```suggestion
              "default": true,
```

File: perception/detected_object_validation/schema/object_position_filter.schema.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          "type": "number",
+          "default": -50.0,
+          "description": "Bound for filtering. Only used if filter_by_xy_position is true"
+        },
+        "filter_target_label": {
+          "type": "object",
+          "properties": {
+            "UNKNOWN": {
+              "type": "boolean",
+              "default": false,
```
Comment (badai-nguyen) [inline perception/detected_object_validation/schema/object_position_filter.schema.json:34]: The same comment.
```suggestion
              "default": true,
```

Comment (batuhanbeytekin) [reply perception/detected_object_validation/schema/object_position_filter.schema.json:34]: The same goes for the correction here.

**Comment by mitsudome-r** [comment]
@batuhanbeytekin It seems like there is a conflict with main branch. Could you resolve the conflicts?

**Comment by batuhanbeytekin** [comment]
> @batuhanbeytekin It seems like there is a conflict with main branch. Could you resolve the conflicts?

Hi @mitsudome-r, my branch is up to date. I think the conflict has been resolved now. Thank you for your reply.

File: perception/detected_object_validation/launch/object_position_filter.launch.xml [hunk 3]
```diff
@@ -1,12 +1,11 @@
-<?xml version="1.0"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <launch>
+  <arg name="object_position_filter_param_file" default="$(find-pkg-share detected_object_validation)/config/object_position_filter.param.yaml"/>
```
Comment (mitsudome-r) [inline perception/detected_object_validation/launch/object_position_filter.launch.xml:3]: Are there any reasons for changing the argument name? Is it to match the name with parameter file name?
Since these arguments could be used by other launch files, it might be better to keep it the same for now. We could fix them when we refactor launch files after parameter work is done. (The same goes for other launch argument names)

**Review by Shin-kyoto — CHANGES_REQUESTED** [review]
@batuhanbeytekin
Please resolve conflicts.

File: perception/autoware_detected_object_validation/launch/obstacle_pointcloud_based_validator.launch.xml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Shin-kyoto) [inline perception/autoware_detected_object_validation/launch/obstacle_pointcloud_based_validator.launch.xml:1]: @batuhanbeytekin 
Why did you remove `encoding="UTF-8"` ?

**Review by YoshiRi — COMMENTED** [review]
Same comments with tomie-san.
Also, please merge [autoware_launch changes](https://github.com/autowarefoundation/autoware_launch/pull/1126) before this PR.

File: perception/autoware_detected_object_validation/launch/obstacle_pointcloud_based_validator.launch.xml (hunk 4 above)
Comment (batuhanbeytekin) [reply perception/autoware_detected_object_validation/launch/obstacle_pointcloud_based_validator.launch.xml]: I now understand it correctly. I apologize for the confusion.

**Comment by Shin-kyoto** [comment]
@YoshiRi 

Can you review this PR, again?

**Comment by Shin-kyoto** [comment]
@YoshiRi 

friendly ping

File: perception/autoware_detected_object_validation/README.md [hunk 5]
```diff
@@ -10,3 +10,21 @@ The purpose of this package is to eliminate obvious false positives of DetectedO
 - [Occupancy grid based validator](occupancy-grid-based-validator.md)
 - [Object lanelet filter](object-lanelet-filter.md)
 - [Object position filter](object-position-filter.md)
+
+### Node Parameters
+
+#### object_lanelet_filter
+
+{{ json_to_markdown("perception/detected_object_validation/schema/object_lanelet_filter.schema.json") }}
```
Comment (mitsudome-r) [inline perception/autoware_detected_object_validation/README.md:18]:
```suggestion
{{ json_to_markdown("perception/autoware_detected_object_validation/schema/object_lanelet_filter.schema.json") }}
```

File: perception/autoware_detected_object_validation/README.md [hunk 6]
```diff
@@ -10,3 +10,21 @@ The purpose of this package is to eliminate obvious false positives of DetectedO
...
+
+#### object_position_filter
+
+{{ json_to_markdown("perception/detected_object_validation/schema/object_position_filter.schema.json") }}
```
Comment (mitsudome-r) [inline perception/autoware_detected_object_validation/README.md:22]:
```suggestion
{{ json_to_markdown("perception/autoware_detected_object_validation/schema/object_position_filter.schema.json") }}
```

File: perception/autoware_detected_object_validation/README.md [hunk 7]
```diff
@@ -10,3 +10,21 @@ The purpose of this package is to eliminate obvious false positives of DetectedO
...
+
+#### obstacle_pointcloud_based_validator
+
+{{ json_to_markdown("perception/detected_object_validation/schema/obstacle_pointcloud_based_validator.schema.json") }}
```
Comment (mitsudome-r) [inline perception/autoware_detected_object_validation/README.md:26]:
```suggestion
{{ json_to_markdown("perception/autoware_detected_object_validation/schema/obstacle_pointcloud_based_validator.schema.json") }}
```

File: perception/autoware_detected_object_validation/README.md [hunk 8]
```diff
@@ -10,3 +10,21 @@ The purpose of this package is to eliminate obvious false positives of DetectedO
...
+
+#### occupancy_grid_based_validator
+
+{{ json_to_markdown("perception/detected_object_validation/schema/occupancy_grid_based_validator.schema.json") }}
```
Comment (mitsudome-r) [inline perception/autoware_detected_object_validation/README.md:30]:
```suggestion
{{ json_to_markdown("perception/autoware_detected_object_validation/schema/occupancy_grid_based_validator.schema.json") }}
```

**Comment by Shin-kyoto** [comment]
@batuhanbeytekin 
Can you rebase the commits not related to your PR?

**Comment by Shin-kyoto** [comment]
@MasatoSaeki 
Can you review this PR?

File: perception/autoware_detected_object_validation/schema/obstacle_pointcloud_based_validator.schema.json [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+          },
+          "default": [10, 10, 10, 10, 10, 10, 10, 10],
+          "description": "The max number of obstacle point clouds in DetectedObjects"
+        },
+        "min_points_and_distance_ratio": {
+          "type": "array",
+          "items": {
+            "type": "integer"
+          },
+          "default": [800, 800, 800, 800, 800, 800, 800, 800],
```
Comment (MasatoSaeki) [inline perception/autoware_detected_object_validation/schema/obstacle_pointcloud_based_validator.schema.json:30]:
```suggestion
            "type": "number"
          },
          "default": [800.0, 800.0, 800.0, 800.0, 800.0, 800.0, 800.0, 800.0],
```
In source code, `min_points_and_distance_ratio` declares as `double`, so we need to follow that type.

**Review by MasatoSaeki — COMMENTED** [review]
@batuhanbeytekin 
Thank you for your PR. I cannot see [Document](https://autowarefoundation.github.io/autoware.universe/pr-7750/perception/autoware_detected_object_validation/) caused by lack of `autoware_` prefix in `README.md`(there are already suggestions), so please modified this.

File: perception/autoware_detected_object_validation/schema/obstacle_pointcloud_based_validator.schema.json [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+          },
+          "default": [10, 10, 10, 10, 10, 10, 10, 10],
+          "description": "The max number of obstacle point clouds in DetectedObjects"
+        },
+        "min_points_and_distance_ratio": {
+          "type": "array",
+          "items": {
+            "type": "number"
+          },
+          "default": [800, 800, 800, 800, 800, 800, 800, 800],
```
Comment (MasatoSaeki) [inline perception/autoware_detected_object_validation/schema/obstacle_pointcloud_based_validator.schema.json:30]:
```suggestion
          "default": [800.0, 800.0, 800.0, 800.0, 800.0, 800.0, 800.0, 800.0],
```

**Review by YoshiRi — APPROVED** [review]
LGTM with [evaluator check](https://evaluation.ci.tier4.jp/evaluation/reports/82d943bc-fcbd-5caa-978b-f442584bc1bd?project_id=prd_jt).
Documentation problem can be solved with another PR.

**Review by Shin-kyoto — APPROVED** [review]

**Comment by MasatoSaeki** [comment]
@batuhanbeytekin 
If you want to capture progress of main branch, you can use `update branch` button. It is easy to do that and. (It looked like you made mistakes when rebase)
And you do not need to assign reviewer who is not related. It only confuses related people.
image
